src/main/java/warehouse/Warehouse.java

```diff
@@ -93,8 +93,7 @@ public double calculateAveragePurchasePriceByDeviceType(DeviceType deviceType) {
             }
         }
         if (numOfDevices == 0) {
-            System.out.println("there aren't devices of kind: " + deviceType);
-            return 0.0;
+            throw new EmptyFilterException("there aren't devices of kind: " + deviceType);
```

Comment:
Il nome dell'eccezione non è molto significativa del risultato che ti aspetti. Con questo nome mi viene da intuire che l'eccezione sia dovuta ad un filtro vuoto. In questo caso forse puoi riutilizzare quella già nominata NoDevicesMatchFiltersCriteriaException, oppure farne una che sia più chiara per l'utente riguardo a che problema si sta verificando.

Reply:
Ma in effetti l'eccezione che lancio qui è dovuta a un filtro vuoto, dovuto al fatto che non sono stati trovati devices compatibili. Comunque va bene, userò l'eccezione che hai detto tu, quando ho corretto il metodo non era ancora stata creata. Per quanto riguarda il link mea culp, rimedio ora